### PR TITLE
Use protocol in pika credentials

### PR DESCRIPTION
**Pull request by kbaikov**
Used pika source as reference: https://github.com/pika/pika/blob/96a92379346285a8f53dd0b3f76cb26f45baf350/pika/credentials.py#L25

### CODE REVIEW

File: stubs/pika/pika/credentials.pyi [hunk 1]
```diff
@@ -1,33 +1,26 @@
...
     TYPE: ClassVar[str]
+    erase_on_connect: bool
     username: str
     password: str
-    erase_on_connect: bool
     def __init__(self, username: str, password: str, erase_on_connect: bool = False) -> None: ...
-    def __eq__(self, other: object) -> bool: ...
-    def __ne__(self, other: object) -> bool: ...
-    def response_for(self, start: Connection.Start) -> tuple[str | None, bytes | None]: ...
-    def erase_credentials(self) -> None: ...
```
Comment (srittau) [inline stubs/pika/pika/credentials.pyi:20]: Why did you remove these methods?

Comment (kbaikov) [reply stubs/pika/pika/credentials.pyi]: my mistake

**Review by srittau — APPROVED** [review]
Thanks!
